controllers/controllers/workloads/cfspace_controller.go

```diff
@@ -150,34 +153,78 @@ func (r *CFSpaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
-func (r *CFSpaceReconciler) createServiceAccounts(ctx context.Context, namespace string) error {
-	err := r.createServiceAccountIfMissing(ctx, &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      kpackServiceAccountName,
-			Namespace: namespace,
-		},
-		ImagePullSecrets: []corev1.LocalObjectReference{
-			{Name: r.packageRegistrySecretName},
-		},
-		Secrets: []corev1.ObjectReference{
-			{Name: r.packageRegistrySecretName},
-		},
+func (r *CFSpaceReconciler) reconcileServiceAccounts(ctx context.Context, space client.Object) error {
+	var (
+		result controllerutil.OperationResult
+		err    error
+	)
+
+	serviceAccounts := new(corev1.ServiceAccountList)
+	err = r.client.List(ctx, serviceAccounts, client.InNamespace(r.rootNamespace))
+	if err != nil {
+		r.log.Error(err, fmt.Sprintf("Error listing service accounts from namespace %s", space.GetNamespace()))
+		return err
+	}
+
+	parentRoleBindingMap := make(map[string]struct{})
```

Comment:
This map is actually a set of reconciled service account names and has nothing to do with role bindings. I would therefore suggest renaming it to `reconciledServiceAccounts` to make that clear.

Reply:
Haha. Can you tell that this was copy/pasted? Thanks.